workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+            listView.onSelectionChanged += OnSelectionChanged;
+            listView.itemsSource = entities.Data;
+
+            var searchField = this.Q<ToolbarSearchField>();
+            searchField.RegisterCallback<ChangeEvent<string>>(OnSearchFieldChanged);
+        }
+
+        public void Update()
+        {
+            if (entitySystem == null)
+            {
+                listView.itemsSource = new List<EntityId>();
+                return;
+            }
+
+            if (entitySystem.ViewVersion != lastViewVersion)
+            {
+                entities.RefreshData(world);
+                listView.itemsSource = entities.Data;
+                lastViewVersion = entitySystem.ViewVersion;
+            }
+        }
+
+        public void SetWorld(World world)
+        {
+            this.world = world;
+            entitySystem = world?.GetExistingSystem<EntitySystem>();
+            lastViewVersion = 0;
+        }
+
+        private void BindItem(VisualElement element, int index)
+        {
+            var entity = entities.Data[index];
+
+            if (!(element is Label label))
+            {
+                return;
+            }
+
+            label.text = entity.ToString();
+        }
+
+        private void OnSelectionChanged(List<object> selections)
+        {
+            // TODO: Error handling
+            // We only allow you to select one at a time
+            var entityData = (EntityData) selections[0];
+            OnEntitySelected?.Invoke(entityData.EntityId);
+        }
+
+        private void OnSearchFieldChanged(ChangeEvent<string> changeEvent)
+        {
+            var searchValue = changeEvent.newValue.Trim();
+            SearchParameters searchParameters;
+            if (long.TryParse(searchValue, out var value))
+            {
+                searchParameters = new SearchParameters { EntityId = new EntityId(value) };
+            }
+            else
+            {
+                searchParameters = new SearchParameters { SearchFragment = searchValue.ToLower() };
+            }
+
+            entities.SearchParameters = searchParameters;
+            entities.Calculate();
+            listView.itemsSource = entities.Data;
+        }
+
+        private struct SearchParameters
+        {
+            public EntityId? EntityId;
+            public string SearchFragment;
+        }
+
+        private class Entities
+        {
+            private static ProfilerMarker refreshDataMarker = new ProfilerMarker("EntityList.RefreshData");
+            private static ProfilerMarker applyFilterMarker = new ProfilerMarker("EntityList.Calculate");
+
+            public List<EntityData> Data { get; } = new List<EntityData>();
+
+            public SearchParameters SearchParameters;
+
+            private List<EntityData> underlying = new List<EntityData>();
+            private World world;
+            private EntityQuery query;
+
+            private readonly EntityQueryDesc queryDesc = new EntityQueryDesc
```

Comment:
This isn't needed. Using `CreateEntityQuery(ComponentType[])` directly will be simpler.

Reply:
Done